### ISSUE TITLE
bug: broken markdown options with Tab size less than 4

### ISSUE THREAD

**Issue by OlegLustenko** [nothingislost/obsidian-codemirror-options#4]
Hi, I really like the last feature "Hide markdown Tokens". It creates for me almost an outliner experience! Thanks for bringing it to the Obsidian community.

I've been playing a little with my Obsidian settings and have noticed that "Hide markdown Token" doesn't display TODOs such as 
[ ], [x] properly with **Tab size(Options -> Editor) less then 4** setting

So steps to reproduce:
**Preconditions**
1) Obisidian codemirror-options is installed 
2) Hide markdown Tokens is enabled

Steps 
1) Create TODO inside Obisidian
2) Set Tab size less than 4

**actual**
Broken layout of tasks
**expected**
Layout stays the same as we have with Tab size 4

Tab Size 2
![Image 2021-09-10 at 11 28 57 AM](https://user-images.githubusercontent.com/11524061/132824648-9e780282-cd12-46b7-a9c2-732879e346eb.jpg)

Tab Size 4
![Image 2021-09-10 at 11 29 33 AM](https://user-images.githubusercontent.com/11524061/132824729-0190d2fd-f9c4-40c7-a09a-7327b1e11b99.jpg)

Real World use-case :)
![Image 2021-09-10 at 11 31 13 AM](https://user-images.githubusercontent.com/11524061/132824939-5ae04234-030d-4273-bba8-96d1d08c8f3b.jpg)

**Have you tried to restart Obisdian after changing Tab size?**
Yes
**Have you tried to remove your theme and disable custom css?**
Yes

**Comment by nothingislost** [comment]
Yeah, unfortunately a lot of things break when changing the tab size from 4 to anything else. It breaks HyperMD's markdown parsing logic. Last I looked, it wasn't an easy fix but I'll take another look.

**Comment by nothingislost** [comment]
One workaround for now is to use spaces rather than tabs. 

![image](https://user-images.githubusercontent.com/89109712/132933551-84662bb7-65e2-49fa-b1e0-ce44ec11d57c.png)

That seems to fix the problem:

![image](https://user-images.githubusercontent.com/89109712/132933547-458011b0-6a4b-4def-97c4-3e618bf59e1c.png)
